### PR TITLE
Add Package JavaDocs everywhere

### PR DESCRIPTION
**Pull request by ST-DDT**
Adds https://github.com/SpongePowered/SpongeAPI/issues/520
Bukkit had a nice overview with a one sentence description of each package.

IMO this is quite useful for a more general overview.

Overview:

Package | Description
--- | -----
[...api](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/package-info.java) | Core classes in the API with general purpose.
[...api.attribute](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/attribute/package-info.java) | Classes to manipulate Attributes such as MAX_HEALTH and MOVEMENT_SPEED.
[...api.block](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/block/package-info.java) | Classes used to manipulate block states of block volumes such as worlds.
[...api.block.tileentity](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/block/tileentity/package-info.java) | Classes that represents special data holder blocks.
[...api.block.tileentity.carrier](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/block/tileentity/carrier/package-info.java) | Classes that represents special data holder blocks with inventories.
[...api.data](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/package-info.java) | Classes used to store and manipulate data on a wide variety of data holders, replaces the insecure NBT access.
[...api.data.manipulator](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/manipulator/package-info.java) | Data classes that can be used to add, alter, copy and delete data from DataHolders.
[...api.data.manipulator.block](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/manipulator/block/package-info.java) | Data classes that can be used to add, alter, copy and delete data from blocks.
[...api.data.manipulator.catalog](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/manipulator/catalog/package-info.java) | Utility classes listing all data classes applicable to a specific part of the API.
[...api.data.manipulator.entity](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/manipulator/entity/package-info.java) | Data classes that can be used to add, alter, copy and delete data from entities.
[...api.data.manipulator.item](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/manipulator/item/package-info.java) | Data classes that can be used to add, alter, copy and delete data from item stacks.
[...api.data.manipulator.tileentity](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/manipulator/tileentity/package-info.java) | Data classes that can be used to add, alter, copy and delete data from tile entities.
[...api.data.property](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/property/package-info.java) | Classes that represent immutable data properties.
[...api.data.type](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/data/type/package-info.java) | Enum like classes used in DataManipulators.
[...api.effect](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/effect/package-info.java) | Classes representing visual or audible effects for viewers.
[...api.effect.particle](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/effect/particle/package-info.java) | Classes representing visual particle effects for viewers.
[...api.effect.sound](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/effect/sound/package-info.java) | Classes representing sound effects for viewers.
[...api.entity](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/package-info.java) | Classes for non-voxel objects that can exist in a world, including all players, monsters, projectiles, etc.
[...api.entity.explosive](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/explosive/package-info.java) | Classes for entities that can explode such as TNT.
[...api.entity.hanging](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/hanging/package-info.java) | Classes for hanging entities such as decorative item frames.
[...api.entity.living](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/living/package-info.java) | Classes for living entities such as players and monsters.
[...api.entity.living.animal](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/living/animal/package-info.java) | Classes for living Animals such as cows and sheeps.
[...api.entity.living.complex](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/living/complex/package-info.java) | Classes for complex entities such as the ender dragon.
[...api.entity.living.golem](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/living/golem/package-info.java) | Classes for golems.
[...api.entity.living.monster](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/living/monster/package-info.java) | Classes for Monsters such as creepers and skeletons.
[...api.entity.player](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/player/package-info.java) | Classes for players.
[...api.entity.player.gamemode](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/player/gamemode/package-info.java) | **Should be moved**
[...api.entity.player.tab](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/player/tab/package-info.java) | Classes for player's tab lists.
[...api.entity.projectile](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/projectile/package-info.java) | Classes for projectiles.
[...api.entity.projectile.explosive](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/projectile/explosive/package-info.java) | Classes for explosive projectiles.
[...api.entity.projectile.explosive.fireball](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/projectile/explosive/fireball/package-info.java) | Classes for fireball projectiles.
[...api.entity.projectile.source](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/projectile/source/package-info.java) | Classes for projectile sources.
[...api.entity.vehicle](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/vehicle/package-info.java) | Classes for vehicles such as boats and Minecarts.
[...api.entity.vehicle.minecart](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/vehicle/minecart/package-info.java) | Classes for minecart.
[...api.entity.weather](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/entity/weather/package-info.java) | Classes for weather related entities.
[...api.event](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/package-info.java) | Classes dedicated to handling triggered code executions.
[...api.event.attribute](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/attribute/package-info.java) | Events related to attributes.
[...api.event.ban](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/ban/package-info.java) | Events related to bans.
[...api.event.block](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/block/package-info.java) | Events related to blocks.
[...api.event.block.tileentity](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/block/tileentity/package-info.java) | Events related to tile entities.
[...api.event.cause](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/cause/package-info.java) | Classes for event causes and reasoning.
[...api.event.cause.reason](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/cause/reason/package-info.java) | Classes for event reasoning.
[...api.event.entity](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/package-info.java) | Events related to entities.
[...api.event.entity.living](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/living/package-info.java) | Events related to living entities.
[...api.event.entity.living.human](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/living/human/package-info.java) | Events related to humans including NPCs and players.
[...api.event.entity.living.human.fishing](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/living/human/fishing/package-info.java) | Events related to fishing humans.
[...api.event.entity.living.player](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/living/player/package-info.java) | **Should be moved**
[...api.event.entity.minecart](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/minecart/package-info.java) | Events related to minecarts.
[...api.event.entity.player](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/player/package-info.java) | Events related to Players excluding NPCs.
[...api.event.entity.player.fishing](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/entity/player/fishing/package-info.java) | Events related to fishing players.
[...api.event.inventory](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/inventory/package-info.java) | Events related to inventories.
[...api.event.message](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/message/package-info.java) | Events related to messages such as commands and chat.
[...api.event.network](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/network/package-info.java) | Events related to network communication.
[...api.event.rcon](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/rcon/package-info.java) | Events related to the remote console access.
[...api.event.server](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/server/package-info.java) | Events related to the Server.
[...api.event.server.query](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/server/query/package-info.java) | Events related to the Server queries.
[...api.event.state](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/state/package-info.java) | Events related to the state of the game.
[...api.event.statistic](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/statistic/package-info.java) | Events related to in game Statistics.
[...api.event.weather](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/weather/package-info.java) | Events related to weather effects.
[...api.event.world](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/event/world/package-info.java) | Events related to the world.
[...api.extra.skylands](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/extra/skylands/package-info.java) | Classes related to the skyland world generation.
[...api.item](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/package-info.java) | Classes related to item stacks and inventories.
[...api.item.inventory](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/package-info.java) | Classes related to inventories.
[...api.item.inventory.crafting](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/crafting/package-info.java) | Classes related to crafting inventories.
[...api.item.inventory.custom](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/custom/package-info.java) | Classes related to custom inventories.
[...api.item.inventory.entity](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/entity/package-info.java) | Classes related to inventories for entities.
[...api.item.inventory.equipment](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/equipment/package-info.java) | Classes related to equipment inventories.
[...api.item.inventory.property](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/property/package-info.java) | Property classes related to inventories.
[...api.item.inventory.slot](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/slot/package-info.java) | Classes for different inventory slot types.
[...api.item.inventory.transaction](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/transaction/package-info.java) | Transaction classes related to inventories.
[...api.item.inventory.type](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/inventory/type/package-info.java) | Classes for different inventory layouts.
[...api.item.merchant](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/merchant/package-info.java) | Classes related to merchants and item trading.
[...api.item.recipe](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/item/recipe/package-info.java) | Classes related to (crafting) recipes.
[...api.network](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/network/package-info.java) | Classes related to network communication.
[...api.plugin](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/plugin/package-info.java) | Classes related to plugins and plugin management.
[...api.potion](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/potion/package-info.java) | Classes related to potion effects.
[...api.resourcepack](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/resourcepack/package-info.java) | Classes related to resource packs.
[...api.scoreboard](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/scoreboard/package-info.java) | Classes related to scorebards.
[...api.scoreboard.critieria](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/scoreboard/critieria/package-info.java) | Classes related to scoreboards defining the update behavior.
[...api.scoreboard.displayslot](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/scoreboard/displayslot/package-info.java) | Classes related to positioning scoreboards.
[...api.scoreboard.objective](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/scoreboard/objective/package-info.java) | Classes related to tracking the actual score value for the entiries in scoreboards.
[...api.scoreboard.objective.displaymode](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/scoreboard/objective/displaymode/package-info.java) | Classes defining the look of the values displayed by scoreboards.
[...api.service](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/package-info.java) | Classes related to service management as well as service definitions.
[...api.service.ban](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/ban/package-info.java) | Service classes related to ban management.
[...api.service.command](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/command/package-info.java) | Service classes related to command management.
[...api.service.config](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/config/package-info.java) | Service classes related to config management.
[...api.service.event](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/event/package-info.java) | Service classes related to event management.
[...api.service.pagination](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/pagination/package-info.java) | Service classes related to pagination handling.
[...api.service.permission](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/permission/package-info.java) | Service classes related to permission management.
[...api.service.permission.context](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/permission/context/package-info.java) | Classes used to separate different permission contexts.
[...api.service.permission.option](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/permission/option/package-info.java) | Permission holder classes more data than just permissions.
[...api.service.persistence](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/persistence/package-info.java) | Classes related to managing and persisting data objects.
[...api.service.profile](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/profile/package-info.java) | Classes related to fetching game profiles.
[...api.service.rcon](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/rcon/package-info.java) | Service classes used to obtain details about the remote console.
[...api.service.scheduler](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/scheduler/package-info.java) | Service classes related to timed code executions.
[...api.service.sql](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/sql/package-info.java) | Service classes related to sql connection management.
[...api.service.user](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/user/package-info.java) | Classes related to fetching user profiles.
[...api.statistic](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/statistic/package-info.java) | Classes related to ingame statistics.
[...api.statistic.achievement](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/statistic/achievement/package-info.java) | Classes related to achievements.
[...api.status](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/status/package-info.java) | Classes related to server status publishing.
[...api.text](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/package-info.java) | Classes related to messaging and text displaying and formating.
[...api.text.action](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/action/package-info.java) | Classes related to actions bound to text parts.
[...api.text.chat](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/chat/package-info.java) | Classes for controlling the type of text being send.
[...api.text.format](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/format/package-info.java) | Classes related to Text formating such as colors and style.
[...api.text.selector](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/selector/package-info.java) | Classes related to selectors.
[...api.text.sink](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/sink/package-info.java) | Classes related to message sinks (message target bundles).
[...api.text.title](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/title/package-info.java) | Classes related to title screens.
[...api.text.translation](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/text/translation/package-info.java) | Text classes related to multi language support.
[...api.util](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/package-info.java) | Multi and single purpose classes to facilitate various programmatic concepts.
[...api.util.annotation](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/annotation/package-info.java) | Utility annotation classes.
[...api.util.ban](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/ban/package-info.java) | Utility classes related to player and user bans.
[...api.util.blockray](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/blockray/package-info.java) | Block ray tracing utility classes.
[...api.util.command](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/command/package-info.java) | Utility classes related to commands.
[...api.util.command.args](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/command/args/package-info.java) | Utility classes related to command arguments.
[...api.util.command.args.parsing](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/command/args/parsing/package-info.java) | Utility classes related to command argument parsing.
[...api.util.command.dispatcher](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/command/dispatcher/package-info.java) | Utility classes related to command dispatching.
[...api.util.command.source](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/command/source/package-info.java) | Utility classes related to command sources.
[...api.util.command.spec](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/command/spec/package-info.java) | Utility classes related to command specifications.
[...api.util.event.callback](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/event/callback/package-info.java) | Utility classes related to event callbacks.
[...api.util.event.factory](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/event/factory/package-info.java) | Utility classes related to event instance/class generation.
[...api.util.event.factory.plugin](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/event/factory/plugin/package-info.java) | Utility classes related to event instance/class generation extensions.
[...api.util.event.superclasses](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/event/superclasses/package-info.java) | Utility class for generated event classes.
[...api.util.reflect](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/reflect/package-info.java) | Classes related to java reflections.
[...api.util.rotation](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/rotation/package-info.java) | Utility classes related to rotations.
[...api.util.weighted](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/util/weighted/package-info.java) | Classes related to weighted randomization.
[...api.world](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/package-info.java) | Classes related to the world and its properties.
[...api.world.biome](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/biome/package-info.java) | Classes related to the world's biomes.
[...api.world.difficulty](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/difficulty/package-info.java) | Classes related to the worlds difficulty.
[...api.world.extent](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/extent/package-info.java) | Classes related to virtually separating worlds into smaller pieces and working with them.
[...api.world.gamerule](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/gamerule/package-info.java) | Classes related to the world's game rules.
[...api.world.gen](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/gen/package-info.java) | Classes related to world generation.
[...api.world.gen.populator](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/gen/populator/package-info.java) | Classes related to the world population (blocks only).
[...api.world.gen.type](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/gen/type/package-info.java) | Classes related to type definitions for the world generator.
[...api.world.storage](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/storage/package-info.java) | Classes related to world storage access.
[...api.world.weather](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/world/weather/package-info.java) | Classes related to the weather in a world.

Please note, that this summary does not contain any links to referenced classes, although the real overview would have them.

If you have any wishes or suggestions please tell me so i can add them here.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/event/entity/living/human/fishing/package-info.java [hunk 1]
```diff
@@ -22,4 +22,10 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
+/**
+ * {@link org.spongepowered.api.event.Event Events} related to
+ * {@link org.spongepowered.api.data.type.Fish fishing humans}.
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/event/entity/living/human/fishing/package-info.java]: I think "humans fishing" would be better.

**Comment by gabizou** [comment]
If I didn't know any better, most of these javadocs are just reiterating what the package name says versus what the package actually represents and how it can be used.

Example:
```java
/**
 * Service classes related to
 * {@link org.spongepowered.api.service.ban.BanService ban management}.
 */
@org.spongepowered.api.util.annotation.NonnullByDefault
package org.spongepowered.api.service.ban;
```
I wonder what *else* could a package named `service.ban` be useful for? Surely it has **something** to do with being a service related to bans... Could you possibly elaborate? Give us a clue as to what can be done with it? How to use it? When to use it?

**Comment by ST-DDT** [comment]
I was asked to not include examples.
https://github.com/SpongePowered/SpongeAPI/issues/520#issuecomment-120238753

**Comment by gabizou** [comment]
> I was asked to not include examples.
> #520 (comment)

Sure, no examples, but not everything is explained in examples. The reasoning for examples not being included is that they can quickly become out of date. Explaining in a general sense of what the packages do in more than a single sentence for some of the packages. Explanation as to why the package is structured as so, etc.

**Comment by ST-DDT** [comment]
Okay, can you give me an example why the ban package is structured that way?
Especially that package is a real myth for me, because everything connected to that service is in an another package (util).

Can you give a short example or outline? I would like to get a feeling of what exactly you do have in mind.

The only reference i had was Bukkit and some other publicly available javadocs. 
Most of them consisted of one or two sentences.

**Comment by ST-DDT** [comment]
@gabizou Something like this?
[org.spongepowered.api.service](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/package-info.java#L27-35)
[org.spongepowered.api.service.ban](https://github.com/ST-DDT/SpongeAPI/blob/JavaDoc/src/main/java/org/spongepowered/api/service/ban/package-info.java#L27-32)

**Comment by simon816** [comment]
@SpongePowered/developers Are we going to go ahead with this?

**Comment by Zidane** [comment]
@simon816 

Yes.

**Comment by ST-DDT** [comment]
Shall I update it to latest master using this PR quality level or will you write them yourself anyway? I cannot start with this until this weekend though.

**Comment by gabizou** [comment]
We'll be going through and documenting things as we ramp up for beta. Having that said, I'd appreciate some smaller javadoc PR's for individual packages/groups so as not to nitpick and prevent one massive addition when it's a problem of smaller additions.

**Comment by gabizou** [comment]
@ST-DDT Mind updating this PR (rebase to eliminate the unnecessary merge commits)?

**Comment by ST-DDT** [comment]
@gabizou Done

**Comment by gabizou** [comment]
While this is a decent attempt, this PR is just a regurgitation of what the package is named. If you really want this to happen, smaller PRs for specific packages can be done with better descriptions explaining what sort of functionalities they pertain to.

**Pull request closed without merging**